shell/platform/darwin/ios/framework/Source/FlutterRestorationPluginTest.mm

```diff
@@ -0,0 +1,193 @@
+  FlutterRestorationPlugin* restorationPlugin =
+      [[FlutterRestorationPlugin alloc] initWithChannel:restorationChannel restorationEnabled:YES];
+  [restorationPlugin restorationComplete];
+
+  NSData* data = [@"testrestortiondata" dataUsingEncoding:NSUTF8StringEncoding];
+  FlutterMethodCall* methodCall = [FlutterMethodCall
+      methodCallWithMethodName:@"put"
+                     arguments:[FlutterStandardTypedData typedDataWithBytes:data]];
+  [restorationPlugin handleMethodCall:methodCall
+                               result:^(id _Nullable result) {
+                                 XCTAssertNil(result);
+                               }];
+  XCTAssertEqual([restorationPlugin restorationData], data);
+}
+
+- (void)testRespondsWithDataSetByFramework {
+  FlutterRestorationPlugin* restorationPlugin =
+      [[FlutterRestorationPlugin alloc] initWithChannel:restorationChannel restorationEnabled:YES];
+  [restorationPlugin restorationComplete];
+
+  NSData* data = [@"testrestortiondata" dataUsingEncoding:NSUTF8StringEncoding];
+  FlutterMethodCall* methodCall = [FlutterMethodCall
+      methodCallWithMethodName:@"put"
+                     arguments:[FlutterStandardTypedData typedDataWithBytes:data]];
+  [restorationPlugin handleMethodCall:methodCall
+                               result:^(id _Nullable result) {
+                                 XCTAssertNil(result);
+                               }];
+  XCTAssertEqual([restorationPlugin restorationData], data);
+
+  __block id capturedResult;
+  methodCall = [FlutterMethodCall methodCallWithMethodName:@"get" arguments:nil];
+  [restorationPlugin handleMethodCall:methodCall
+                               result:^(id _Nullable result) {
+                                 capturedResult = result;
+                               }];
+  XCTAssertEqual([capturedResult count], 2u);
+  XCTAssertEqual([capturedResult objectForKey:@"enabled"], @YES);
+  XCTAssertEqual([[capturedResult objectForKey:@"data"] data], data);
+}
+
+- (void)testResetClearsData {
+  FlutterRestorationPlugin* restorationPlugin =
+      [[FlutterRestorationPlugin alloc] initWithChannel:restorationChannel restorationEnabled:YES];
+  [restorationPlugin restorationComplete];
+
+  NSData* data = [@"testrestortiondata" dataUsingEncoding:NSUTF8StringEncoding];
+  FlutterMethodCall* methodCall = [FlutterMethodCall
+      methodCallWithMethodName:@"put"
+                     arguments:[FlutterStandardTypedData typedDataWithBytes:data]];
+  [restorationPlugin handleMethodCall:methodCall
+                               result:^(id _Nullable result) {
+                                 XCTAssertNil(result);
+                               }];
+  XCTAssertEqual([restorationPlugin restorationData], data);
+
+  [restorationPlugin reset];
+  XCTAssertNil([restorationPlugin restorationData]);
+
+  __block id capturedResult;
+  methodCall = [FlutterMethodCall methodCallWithMethodName:@"get" arguments:nil];
```

Comment:
Optional: It would be nice if these method names where placed as consts in the header so we don't have to duplicate them.  No one should have to refer to them by string literal.